metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/reporters/prometheus/PrometheusMetricsExporter.java

```diff
@@ -213,7 +227,12 @@ private MetricFamilySamples toMetricFamilySamples(PrometheusInstanceSample is) {
                 .map(s -> {
                     String sampleName = name;
 
-                    if (s.hasNameSuffix()) {
+                    if (s.hasName()) {
```

Comment:
Suggest to create dedicated methods

```
private String buildName(MetricName name) {
    final var sanitizeMetricName = sanitizeMetricName(join(NAME_PARTS_DELIMITER, name));
    return convertNameToLowercase ? sanitizeMetricName.toLowerCase(locale) : sanitizeMetricName;
}

private String buildSuffix(PrometheusSample ps) {
    final var suffix = ps.nameSuffix();
    final var sanitizedSuffix = sanitizeMetricName(suffix);
    return convertNameToLowercase ? sanitizedSuffix.toLowerCase(locale) : sanitizedSuffix;
}
```

then line 217 - 220 could be
```
 String name = buildName(is.name());
```

lines 230 - 235:
```
if (s.hasName()) {
    sampleName = buildName(s.name());
} else if (s.hasNameSuffix()) {
    sampleName += buildSuffix(s);
}
```